pkg/reconciler/extensions.go

```diff
@@ -0,0 +1,75 @@
+package reconciler
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/operator-framework/helm-operator-plugins/pkg/extension"
+	"helm.sh/helm/v3/pkg/chartutil"
+	"helm.sh/helm/v3/pkg/release"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+)
+
+type extensions struct {
+	extensions []extension.Extension
+}
+
+func (es *extensions) register(e extension.Extension) {
+	es.extensions = append(es.extensions, e)
+}
+
+func (es *extensions) len() int {
+	return len(es.extensions)
+}
+
+func (es *extensions) get(idx int) extension.Extension {
+	return es.extensions[idx]
+}
+
+func (es *extensions) iterate(f func(e extension.Extension) error) error {
+	var err error
+	for _, e := range es.extensions {
+		err = f(e)
+		if err != nil {
+			return err
+		}
+	}
+	return err
+}
+
+func (es *extensions) loggerInto(l logr.Logger) {
```

Comment:
The method name doesn't make sense. The function below is to be read as "inject logger into", by taking the package name into account. And "into" at the end of a function is commonly read as to refer to the (last) argument of the function call. So if we need it, it should be called `injectLoggerIntoAll`